### ISSUE TITLE
IOU - Error in LHN and requester cant access the IOU if it is sent to a user with no history

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#23987]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the staging webapp as user A
2. Click on + button and initiate a new Chat with user B, which you do not have a conversation history with
3. Send a text message
4. Request money from user B
5. After request is sent, observe the LHN chats >>> (result 1)
6. Click on the error chat that appeared in LHN
7. Go back to the chat with user B and click on the request preview >>> (result 2)

## Expected Result:
Result 1: there should be no error when user A requested money from user B
Result 2: requester (user A) should be able to enter request report page without any issues

## Actual Result:
Result 1: At step 5, an error chat appeared in LHN with red dot, when the error is clicked, user is directed to concierge chat after a second
Result 2: At step 7, when the request preview is clicked, user is directed to a blocker page, with the message "You do not have access to this chat"

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.48.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/481a4b84-fc5a-41d3-a638-9e2fa7e58a6d

![Bug6149147_Screen_Shot_2023-08-01_at_12 26 07_PM](https://github.com/Expensify/App/assets/78819774/2f50d97d-00a5-4c06-a19c-2a77b846a94d)

![Bug6149147_Screen_Shot_2023-08-01_at_12 26 33_PM](https://github.com/Expensify/App/assets/78819774/f903a255-f76b-41e5-bef1-c5c16943c00e)

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause- Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by trjExpensify** [comment]
I think this is a dupe of https://github.com/Expensify/App/issues/22480 with the same RCA related to accountIDs. Happy for a second opinion from @aldo-expensify first before I close it out!

**Comment by trjExpensify** [comment]
Back from OoO today, reached out to Aldo!

**Comment by aldo-expensify** [comment]
Yes, it is the same, we are not sending the email and neither the accountID in the `RequestMoney` command

<img width="793" alt="image" src="https://github.com/Expensify/App/assets/87341702/97246aab-1a67-4beb-96fe-a006e92ad290">

**Comment by trjExpensify** [comment]
Thanks! :)
